### PR TITLE
replace split dependency to split2

### PR DESCRIPTION
**Pull request by jordanSu**
Those files which require split are replaced to split2. The package.json dependencies are also updated.

### CODE REVIEW

File: problems/combiner/problem.md [hunk 1]
```diff
@@ -61,15 +61,15 @@ front 'end' of the stream, reads the associated JSON, processes the input book
 data by grouping it by genre and produces a gzipped result stream from which 
 the result may be read.
 
-As in the previous LINES adventure, the `split` module is very handy here. You
+As in the previous LINES adventure, the `split2` module is very handy here. You
 can put a split stream directly into the stream-combiner pipeline.
 Note that split can send empty lines too.
```
Comment (ccarruitero) [inline problems/combiner/problem.md:66]: I think this `split` refer to the module too, so we should update to `split2`

File: problems/lines/problem.md [hunk 2]
```diff
@@ -40,13 +40,13 @@ $ echo -e 'one\ntwo\nthree' | node split.js
 'three'
 ```
 
-Your own program could use `split` in this way, and you should transform the
+Your own program could use `split2` in this way, and you should transform the
 input and pipe the output through to `process.stdout`. Keep in mind that,
 if you decide to use this technique, `split2` might be actually needed,
 depending on the versions of the other dependencies.
```
Comment (ccarruitero) [inline problems/lines/problem.md:46]: we can remove this sentence (`Keep in mind ...`), since we are recommending use `split2` in all problem.

**Comment by jordanSu** [comment]
@ccarruitero Thanks for the advices.
The problem description is updated.

**Comment by ccarruitero** [comment]
Thanks @jordanSu,

I'll merge now

**Pull request merged**